src/org/parosproxy/paros/db/paros/ParosTableHistory.java

```diff
@@ -402,7 +402,53 @@ private RecordHistory build(ResultSet rs) throws HttpMalformedHeaderException, S
 		return history;
 	
 	}
-	
+
+	private List<Integer> getHistoryIdsByParams(long sessionId, int startAtHistoryId,  boolean includeHistTypes, int... histTypes) throws DatabaseException {
+		try {
+			boolean hasHistTypes = histTypes != null && histTypes.length > 0;
+			int strLength = hasHistTypes ? 97 : 68;
+			StringBuilder strBuilder = new StringBuilder(strLength);
+			strBuilder.append("SELECT ").append(HISTORYID);
+			strBuilder.append(" FROM ").append(TABLE_NAME).append(" WHERE ").append(SESSIONID).append(" = ?");
+			if (hasHistTypes && includeHistTypes) {

```

Comment:
This could have less conditions and avoid duplication:
```Java
if (hasHistTypes) {
    strBuilder.append(" AND ").append(HISTTYPE);
    if (!includeHistTypes) {
        strBuilder.append(" NOT");
    }
    strBuilder.append(" IN ( UNNEST(?) )");
}
```